src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BranchSCMHead.java

```diff
@@ -51,23 +49,10 @@
      * Constructor.
      *
      * @param branchName the branch name
-     * @deprecated use {@link #BranchSCMHead(String, BitbucketRepositoryType)}
      */
-    @Deprecated
-    @Restricted(DoNotUse.class)
     public BranchSCMHead(String branchName) {
```

Comment:
I cross checked this: This function was not used in plugin. Therefore, i think repositoryType can be hard-coded to BitbucketRepositoryType.GIT instead of null.